app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisUseCase.kt

```diff
@@ -0,0 +1,45 @@
+package co.netguru.baby.monitor.client.feature.voiceAnalysis
+
+import co.netguru.baby.monitor.client.common.ISchedulersProvider
+import co.netguru.baby.monitor.client.common.Randomiser
+import co.netguru.baby.monitor.client.data.DataRepository
+import co.netguru.baby.monitor.client.feature.communication.websocket.Message
+import co.netguru.baby.monitor.client.feature.communication.websocket.MessageController
+import io.reactivex.Completable
+import io.reactivex.Single
+import io.reactivex.rxkotlin.subscribeBy
+import timber.log.Timber
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class VoiceAnalysisUseCase @Inject constructor(
+    private val dataRepository: DataRepository,
+    private val schedulersProvider: ISchedulersProvider,
+    private val randomiser: Randomiser
+) {
+    fun chooseVoiceAnalysisOption(
+        messageController: MessageController,
+        voiceAnalysisOption: VoiceAnalysisOption
+    ): Single<Boolean> {
+        val sentMessage = Message(
+            voiceAnalysisOption = voiceAnalysisOption.name,
+            confirmationId = randomiser.getRandomDigits(4).joinToString("")
+        )
+        return Completable.fromAction {
+            messageController.sendMessage(sentMessage)
+        }
+            .andThen(
+                messageController.receivedMessages()
+                    .filter { it.confirmationId == sentMessage.confirmationId }
+                    .timeout(5, TimeUnit.SECONDS, schedulersProvider.io())
+                    .firstOrError()
+                    .flatMap { return@flatMap Single.just(true) })
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/8b49d288c8cda524f66973edfcfe165e7a30812e/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/voiceAnalysis/VoiceAnalysisUseCase.kt#L36"></span>Expression with labels increase complexity and affect maintainability.</td>
    </tr>
  </tbody>
</table>